src/Umbraco.Core/Configuration/Models/TypeFinderSettings.cs

```diff
@@ -22,4 +22,9 @@ public class TypeFinderSettings
     ///     scanning for plugins based on different root referenced assemblies you can add the assembly name to this list.
     /// </summary>
     public IEnumerable<string>? AdditionalEntryAssemblies { get; set; }
+
+    /// <summary>
+    ///     Gets or sets a value for the assemblies that will be excluded from scanning.
+    /// </summary>
+    public string[] AdditionalAssemblyExclusionEntries { get; set; } = null!;
```

Comment:
Note that using arrays in configuration settings will [bind them using the array indices](https://learn.microsoft.com/en-gb/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0#bind-an-array), so you can't (easily) add or remove items when [multiple configuration sources](https://learn.microsoft.com/en-gb/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0#naming-of-environment-variables) get merged.

For example, if `appsettings.json` contained:
```json
{
    "TypeFinder": {
        "AdditionalAssemblyExclusionEntries": [ "WindowsBase", "PresentationFramework", "ReachFramework", "PresentationCore" ]
      }
}
```

And `appsettings.Development.json`:
```json
{
    "TypeFinder": {
        "AdditionalAssemblyExclusionEntries": [ "MyAdditionalDevelopmentAssembly" ]
      }
}
```

The actual setting value will be:
1. `MyAdditionalDevelopmentAssembly`
2. `PresentationFramework`
3. `ReachFramework`
4. `PresentationCore`

The `WindowsBase` value is overwritten by array index and the other values aren't removed...

Reply:
Fair point Ronald, but IMO this is the lesser evil :) 

Ideally, this would just have been an IOptions that was only configurable by code, as I would not expect this configuration to change between environments.

IOptions is sadly not available for the TypeFinder as this is part of UmbracoBuilder.